### PR TITLE
Make app work gracefully when location access is disabled

### PR DESCRIPTION
**Pull request by jamietsao**
https://github.com/jamietsao/codepath-mapstagram/issues/83

### CODE REVIEW

File: Mapstagram/Services/LocationService.swift [hunk 1]
```diff
@@ -36,7 +36,11 @@ class LocationService : NSObject, CLLocationManagerDelegate {
         locationManager.delegate = self
     }
     
-    func startService(presentingViewController: UIViewController, onNewLocation: (CLLocation) -> (), onLocationFailure: (NSError) -> ()) {
+    func isLocationAuthorized() -> Bool {
```
Comment (jamietsao) [inline Mapstagram/Services/LocationService.swift]: Added new helper method that returns whether or not the user has disabled location access

File: Mapstagram/Services/LocationService.swift [hunk 2]
```diff
@@ -36,7 +36,11 @@ class LocationService : NSObject, CLLocationManagerDelegate {
...
+        return CLLocationManager.authorizationStatus() == .AuthorizedWhenInUse || CLLocationManager.authorizationStatus() == .AuthorizedAlways
+    }
+    
+    func startService(presentingViewController: UIViewController, onNewLocation: (CLLocation) -> (), onLocationFailure: (NSError) -> (), onPermissionCancel: () -> ()) {
```
Comment (jamietsao) [inline Mapstagram/Services/LocationService.swift]: Added additional handler for the case where the user hits cancel on the UIAlert

File: Mapstagram/Services/LocationService.swift [hunk 3]
```diff
@@ -60,7 +64,10 @@ class LocationService : NSObject, CLLocationManagerDelegate {
                 message: "In order to be see Instagram photos in your location, please open this app's settings and set location access to 'While Using the App'.",
                 preferredStyle: .Alert)
             
-            let cancelAction = UIAlertAction(title: "Cancel", style: .Cancel, handler: nil)
+            let cancelAction = UIAlertAction(title: "Cancel", style: .Cancel) {
+                (action) in
+                onPermissionCancel()
```
Comment (jamietsao) [inline Mapstagram/Services/LocationService.swift]: Invoke new handler for when user clicks cancel

File: Mapstagram/Services/SessionService.swift [hunk 4]
```diff
@@ -42,8 +42,9 @@ class SessionService {
         
         static let distancesArray = [oneBlock, twoBlocks, oneMile, twoMiles, threeMiles]
     }
-    
-    private var searchLocation: Location!
+
+    // default to SF for cases where location access hasn't been granted
+    private var searchLocation: Location = Location(name: "San Francisco", lat: 37.788022, long: -122.399797)
```
Comment (jamietsao) [inline Mapstagram/Services/SessionService.swift]: Added default of SF to SessionService, which is used on the initial load of the app if location access is disabled and user clicks cancel

File: Mapstagram/ViewControllers/MapViewController.swift [hunk 5]
```diff
@@ -71,10 +71,28 @@ class MapViewController: MediaViewController {
...
+                    return
+                }
             }
         )
     }
     
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        if LocationService.getInstance().isLocationAuthorized() {
```
Comment (jamietsao) [inline Mapstagram/ViewControllers/MapViewController.swift]: Show or hide the currentLocationButton on the map

**Comment by kristen** [comment]
Looks good. Merge it.

**Pull request merged**